src/stash/src/postgres.rs

```diff
@@ -1464,10 +1464,17 @@ impl DebugStashFactory {
 
 impl Drop for DebugStashFactory {
     fn drop(&mut self) {
-        assert!(
-            self.dropped,
-            "You forgot to call `drop()` on a `DebugStashFactory` before dropping it! You may also \
-            see this if a test panicked before calling `drop()`."
-        );
+        let message =
+            "You forgot to call `drop()` on a `DebugStashFactory` before dropping it! You \
+        may also see this if a test panicked before calling `drop()`.";
+
+        if !self.dropped {
+            // Don't double panic so we keep the stack trace relatively small.
+            if std::thread::panicking() {
+                tracing::error!("{message}");
+            } else {
+                panic!("{message}");
+            }
+        }
```

Comment:
Should we update the `You may also see this if a test panicked before calling drop().` part of the message since now we now for sure if the test is panicking or not?

Reply:
At first I got rid of that part of the message, but then re-added it because I wanted to leave some indication to the developer that the `DebugStashFactory` getting dropped wasn't the actual issue. For example, if your test fails and all you see is 
```
"You forgot to call `drop()` on a `DebugStashFactory` before dropping it!"
```
Then the developer might miss the actual error and go down the wrong path.

I could also be _waaaay_ over thinking this, so I'll go with whatever you want :)

Reply:
No I think it's fine as is.